src/extensions/default/WebPlatformDocs/main.js

```diff
@@ -29,98 +29,135 @@ define(function (require, exports, module) {
+            var path = ExtensionUtils.getModulePath(module, fileName),
                 file = FileSystem.getFileForPath(path);
-            
+
             FileUtils.readAsText(file)
                 .done(function (text) {
                     var jsonData;
                     try {
                         jsonData = JSON.parse(text);
                     } catch (ex) {
-                        console.error("Malformed CSS documentation database: ", ex);
+                        console.error("Malformed documentation database: ", path, ex);
                         result.reject();
                     }
                     result.resolve(jsonData);  // ignored if we already reject()ed above
                 })
                 .fail(function (err) {
-                    console.error("Unable to load CSS documentation database: ", err);
+                    console.error("Unable to load documentation database: ", path,  err);
                     result.reject();
                 });
-            
-            _cssDocsPromise = result.promise();
+
+            promiseCache[fileName] = result.promise();
         }
-        
-        return _cssDocsPromise;
+
+        return promiseCache[fileName];
     }
-    
-    
+
+
     /**
-     * Inline docs provider. Currently looks up docs based on CSS properties only.
+     * Inline docs provider.
      *
      * @param {!Editor} editor
      * @param {!{line:Number, ch:Number}} pos
      * @return {?$.Promise} resolved with an InlineWidget; null if we're not going to provide anything
      */
     function inlineProvider(hostEditor, pos) {
-        var langId = hostEditor.getLanguageForSelection().getId();
-        // Only provide docs when cursor is in CSS content
-        if (langId !== "css" && langId !== "scss" && langId !== "less") {
+        var jsonFile, propInfo,
+            propQueue = [], // priority queue of propNames to try
+            langId = hostEditor.getLanguageForSelection().getId(),
+            supportedLangs = ["css", "scss", "less", "html"],
+            langIndex = langId ? supportedLangs.indexOf(langId) : -1; // fail if langId is falsy
+
+        // Only provide docs when cursor is in supported language
+        if (langIndex < 0) {
             return null;
         }
-        
+
         // Only provide docs if the selection is within a single line
         var sel = hostEditor.getSelection();
         if (sel.start.line !== sel.end.line) {
             return null;
         }
-        
-        // Explicitly use selection start rather than pos, which is usually selection end
-        var cssInfo = CSSUtils.getInfoAtPos(hostEditor, sel.start);
-        
-        // Are we at a propety name (or in a value where name is discernible?)
-        if (cssInfo && cssInfo.name) {
-            var cssPropName = cssInfo.name,
-                result = new $.Deferred();
-            
+
+        if (langIndex <= 2) { // CSS-like language
```

Comment:
Rather than comparing index, can you check index for specific variable mentioned above i.e. cssSupportedLangs and htmlSupportedLangs